beekeeper-api/src/test/java/com/expediagroup/beekeeper/api/BeekeeperControllerTest.java

```diff
@@ -0,0 +1,120 @@
+import static com.expediagroup.beekeeper.core.model.HousekeepingStatus.SCHEDULED;
+import static com.expediagroup.beekeeper.core.model.LifecycleEventType.EXPIRED;
+
+import java.time.Duration;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageImpl;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.jpa.domain.Specification;
+import org.springframework.test.web.servlet.result.MockMvcResultHandlers;
+import org.springframework.test.web.servlet.MockMvc;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import com.expediagroup.beekeeper.core.model.HousekeepingEntity;
+import com.expediagroup.beekeeper.core.model.HousekeepingMetadata;
+import com.expediagroup.beekeeper.core.repository.HousekeepingMetadataRepository;
+
+@ExtendWith(MockitoExtension.class)
+public class BeekeeperControllerTest {
+
+  @Autowired
+  private MockMvc mockMvc;
+  @Autowired
+  private ObjectMapper objectMapper;
+
+  @Mock
+  private Specification<HousekeepingMetadata> spec;
+  @Mock
+  private Pageable pageable;
+  @Mock
+  private HousekeepingEntityService housekeepingEntityService;
+
+  private HousekeepingMetadata table1;
+  private HousekeepingMetadata table2;
+
+  @BeforeEach
+  public void createTables(){
+
+    LocalDateTime CREATION_TIMESTAMP = LocalDateTime.now(ZoneId.of("UTC"));
+
+    table1 = new HousekeepingMetadata.Builder()
+        .path("s3://some/path/")
+        .databaseName("aRandomDatabase")
+        .tableName("aRandomTable")
+        .partitionName("event_date=2020-01-01/event_hour=0/event_type=A")
+        .housekeepingStatus(SCHEDULED)
+        .creationTimestamp(CREATION_TIMESTAMP)
+        .modifiedTimestamp(CREATION_TIMESTAMP)
+        .cleanupDelay(Duration.parse("P3D"))
+        .cleanupAttempts(0)
+        .lifecycleType(EXPIRED.toString())
+        .build();
+    table2 = new HousekeepingMetadata.Builder()
+        .path("s3://some/path2/")
+        .databaseName("aRandomDatabase2")
+        .tableName("aRandomTable2")
+        .partitionName("event_date=2020-01-012/event_hour=2/event_type=B")
+        .housekeepingStatus(DELETED)
+        .creationTimestamp(CREATION_TIMESTAMP)
+        .modifiedTimestamp(CREATION_TIMESTAMP)
+        .cleanupDelay(Duration.parse("P3D"))
+        .cleanupAttempts(0)
+        .lifecycleType(EXPIRED.toString())
+        .build();
+  }
+  
+  @Test
+  public void getAllTables() throws Exception {
```

Comment:
Your tests in this class might get long and they might test many cases so maybe pick a [good naming convention](https://dzone.com/articles/7-popular-unit-test-naming) from here and stick with that